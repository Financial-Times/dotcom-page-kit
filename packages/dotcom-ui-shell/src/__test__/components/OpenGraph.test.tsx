import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/OpenGraph'

describe('dotcom-ui-shell/src/components/OpenGraph', () => {
  it('renders the given Open Graph data to meta tags', () => {
    const fixture = {
      og: {
        title: 'Open Graph title',
        video: 'https://my.site/video.mp4',
        'video:type': 'video/mp4',
        'video:width': 640,
        'video:height': 360
      },
      article: {
        published_time: '2019-02-27T06:59:35.000Z',
        modified_time: '2019-02-27T16:01:21.000Z',
        author: ['Joe Bloggs', 'Jane Doe']
      }
    }

    const tree = renderer.create(<Subject openGraph={fixture} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
