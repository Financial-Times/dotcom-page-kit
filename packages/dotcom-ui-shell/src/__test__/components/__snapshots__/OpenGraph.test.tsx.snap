// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-shell/src/components/OpenGraph renders the given Open Graph data to meta tags 1`] = `
Array [
  <meta
    content="Open Graph title"
    property="og:title"
  />,
  <meta
    content="https://my.site/video.mp4"
    property="og:video"
  />,
  <meta
    content="video/mp4"
    property="og:video:type"
  />,
  <meta
    content={640}
    property="og:video:width"
  />,
  <meta
    content={360}
    property="og:video:height"
  />,
  <meta
    content="2019-02-27T06:59:35.000Z"
    property="article:published_time"
  />,
  <meta
    content="2019-02-27T16:01:21.000Z"
    property="article:modified_time"
  />,
  <meta
    content="Joe Bloggs"
    property="article:author"
  />,
  <meta
    content="Jane Doe"
    property="article:author"
  />,
]
`;
